contentcuration/contentcuration/management/commands/mark_incomplete.py

```diff
@@ -11,62 +11,64 @@
 from contentcuration.models import AssessmentItem
 from contentcuration.models import ContentNode
 from contentcuration.models import File
+from contentcuration.models import License
 
 logmodule.basicConfig(level=logmodule.INFO)
 logging = logmodule.getLogger('command')
 
 
-CHUNKSIZE = 1000
+CHUNKSIZE = 10000
 
 
 class Command(BaseCommand):
 
-    def mark_complete_field(self, query, complete=False):
-        i = 0
-        count = 0
-        node_ids = query[i:i + CHUNKSIZE]
-        while node_ids:
-            ContentNode.objects.filter(pk__in=node_ids).update(complete=complete)
-            i += CHUNKSIZE
-            count += len(node_ids)
-            node_ids = query[i:i + CHUNKSIZE]
-        return count
+    def add_arguments(self, parser):
+        parser.add_argument("--chunksize", dest="chunksize", default=CHUNKSIZE)
 
     def handle(self, *args, **options):
+        chunksize = options["chunksize"]
+
         start = time.time()
 
-        # Mark invalid topics
-        topicstart = time.time()
-        logging.info('Marking topics...')
-        query = ContentNode.objects.filter(kind_id=content_kinds.TOPIC, title='', complete__isnull=True).values_list('id', flat=True)
-        count = self.mark_complete_field(query)
-        logging.info('Marked {} invalid topics (finished in {})'.format(count, time.time() - topicstart))
+        # Mark invalid titles
+        titlestart = time.time()
+        logging.info('Marking blank titles...')
+        count = ContentNode.objects.filter(title='').order_by().update(complete=False)
+        logging.info('Marked {} invalid titles (finished in {})'.format(count, time.time() - titlestart))
+
+        # Mark invalid licenses
+        licensestart = time.time()
+        logging.info('Marking blank licenses...')
+        count = ContentNode.objects.exclude(kind_id=content_kinds.TOPIC).filter(license__isnull=True).order_by().update(complete=False)
```

Comment:
Since some might have both the title and license problem, could you use `filter(complete=False)` here and in other queries?